libs/community/langchain_community/vectorstores/cloudSQL.py

```diff
@@ -0,0 +1,330 @@
+            raise ValueError("Both metadata_columns and ignore_metadata_columns have been provided.")
+
+        if self.engine.dialect.has_table(self.engine.connect(), self.table_name):
+            inspector = inspect(self.engine)
+            column_names = [column['name'] for column in inspector.get_columns(self.table_name)]
+            if self.embedding_column not in column_names:
+                raise ValueError(f"Column {self.embedding_column} does not exist")
+            if self.content_column not in column_names:
+                raise ValueError(f"Column {self.content_column} does not exist")
+            if 'metadata' not in column_names:
+                raise ValueError(f"Column 'metadata' does not exist")
+            if 'uuid' not in column_names:
+                raise ValueError(f"Column 'uuid' does not exist")
+
+        self.create_default_table(metadata)
+
+    def create_vector_extension(self):
+        """Creates the vector extsion to the specified database."""
+        query = sqlalchemy.text("CREATE EXTENSION IF NOT EXISTS vector")
+        with self.engine.connect() as connection:
+            connection.execute(query)
+
+    def create_default_table(self, metadata):
+        """Creates the default table."""
+        table = Table(
+            self.table_name, metadata,
+            Column('uuid', UUID(as_uuid=True), primary_key=True, default=uuid.uuid4),
+            Column('content', String, nullable=False),
+            Column('embedding', Vector(), nullable=True),
+            Column('metadata', JSON, nullable=True)
+            )
+
+        metadata.create_all(self.engine)
+
+    
+    @classmethod
+    def init_vector_store(
+        cls, 
+        engine: sqlalchemy.engine.Engine, 
+        table_name: str, 
+        vector_size: int, 
+        content_column: str='content', 
+        embedding_column: str='embedding', 
+        metadata_columns: Optional[str, List[str]]='metadata',
+        overwrite_existing: bool=False, 
+        store_metadata: bool=False
+    ) -> None:
+
+        """Creating a non-default vectorstore table"""
+
+        metadata = sqlalchemy.MetaData()
+
+        table = Table(
+            table_name, metadata,
+            Column('uuid', UUID(as_uuid=True), primary_key=True, default=uuid.uuid4),
+            Column(content_column, String, nullable=True),
+            Column(embedding_column, Vector(vector_size), nullable=True),
+            Column(metadata_columns, JSON, nullable=True),
+
+        )
+
+        metadata.create_all(engine)
+    
+    @classmethod
+    def from_documents(
+        cls, 
+        docs: List[Document], 
+        engine: sqlalchemy.engine.Engine, 
+        table_name: str,
+        embedding_service: Embeddings
+    ):
+         """Return VectorStore initialized from documents and embeddings."""
+            
+        texts = [d.page_content for d in docs]
+        metadatas = [d.metadata for d in docs]
```

Comment:
nit: I think this should probably be the other way around -- e.i. from_texts should wrap the texts in a doc and call from_documents. This way there is only 2 copy of the data vs 3